src/dwarf.rs

```diff
@@ -327,6 +329,70 @@ mod obj {
     }
 }
 
+#[cfg(target_os="macos")]
+mod obj {
+    extern crate gimli;
+    extern crate libproc;
+    extern crate object;
```

Comment:
Seems `object` is something the gimli folks wrote to make cross-platform stuff easier. It should also work for elf files on linux. If you just `#[cfg]` the method of getting the path, then I think the other functions you've written here should work on both platforms. That's a cool cleanup we can do later. There's also an opportunity to replace reading with memory mapping, I think. Nice :-)